carbon-design-system/carbon
# fix(icon-build-helpers): update parser to replace \t sequences

Closes #8811

This PR updates our builder to replace tab characters `\t` with a space character. This issue seemed to always be at play in our builder since the AI SVG files that are parsed contain a `\t` character in them. In previous iterations of the builder, it would embed this character in the final React component. In our latest builder, it was directly embedding the `\t` character causing errors to occur in #8648 

With this update, all SVG paths should be correct. To verify, follow the testing/reviewing steps below. If this change is successful, a follow-up PR will be made for adding the next icons into the codebase.

#### Changelog

**New**

**Changed**

- Update our SVG parser in `icon-build-helpers` to replace `\t` characters with space
- Refactor the parser to control the emitted format to make migration simpler in the future if we need to switch parsing libraries

**Removed**

#### Testing / Reviewing

- Run `yarn lerna run build --scope='@carbon/icons-react' --include-dependencies` after pulling down the PR
- Run `cd codemods` to change into the codemods directory
- Run `yarn jscodeshift --parser babylon -t transforms/icons-react-size-prop.js '../packages/react/src/components'` to apply the codemod for icons-react to all of our icons
- Go through the list of components Andrew cited in #8648 and verify the icons render as expected, they are roughly:
  - [ ] TreeView
  - [ ] Password Input
  - [ ] DatePicker Input
